### PR TITLE
pgcli 0.10.2

### PR DESCRIPTION
**Pull request by tvon**
http://pgcli.com/

The test is admittedly not very good, but since we don't know how to connect to the local postgres installation I'm not sure what other kind of test can be relied upon.

### CODE REVIEW

File: Library/Formula/pgcli.rb [hunk 1]
```diff
@@ -0,0 +1,74 @@
+PYLIBS = {
```
Comment (tdsmith) [inline Library/Formula/pgcli.rb]: These should be written in terms of explicit `resource`s, alas.

File: Library/Formula/pgcli.rb [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+  },
+}
+
+class Pgcli < Formula
+  homepage "http://pgcli.com/"
+  url "https://pypi.python.org/packages/source/p/pgcli/pgcli-0.10.2.tar.gz"
+  sha1 "c34330335e08418aba1c406ac846b37daaf21262"
+
+  depends_on :python if MacOS.version <= :snow_leopard
+  depends_on "openssl"
```
Comment (tdsmith) [inline Library/Formula/pgcli.rb]: Does pgcli or one of its dependents link directly against openssl?

Comment (tvon) [inline Library/Formula/pgcli.rb]: I believe it's `psycopg2` that requires openssl.

Comment (tdsmith) [inline Library/Formula/pgcli.rb]: Cool, thanks.

File: Library/Formula/pgcli.rb [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+    ENV.prepend_create_path "PYTHONPATH", libexec/"lib/python2.7/site-packages"
+    system "python", *Language::Python.setup_install_args(libexec)
+
+    bin.install Dir["#{libexec}/bin/*"]
+    bin.env_script_all_files(libexec/"bin", :PYTHONPATH => ENV["PYTHONPATH"])
+  end
+
+  test do
+    system "pgcli", "--help"
```
Comment (tdsmith) [inline Library/Formula/pgcli.rb]: Use `bin/"pgcli"`

File: Library/Formula/pgcli.rb [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+  sha1 "c34330335e08418aba1c406ac846b37daaf21262"
+
+  depends_on :python if MacOS.version <= :snow_leopard
+  depends_on "openssl"
+  depends_on :postgresql
+
+  resource "click" do
+    url "https://pypi.python.org/packages/source/c/click/click-3.3.tar.gz"
+    sha1 "d716a932b930d71059e49465b6b42e833808369a"
+  end
```
Comment (tdsmith) [inline Library/Formula/pgcli.rb]: Thanks for changing this; can you put an additional newline between each `resource` block?

**Comment by tdsmith** [comment]
This is looking good aside from style nits; thanks for submitting!

File: Library/Formula/pgcli.rb (hunk 3 above)
Comment (mikemcquaid) [inline Library/Formula/pgcli.rb]: Can this test be modified to do something more substantial than e.g. `--version` or `--help`? See `cmake.rb` for an example of an application formula with a good test and `tinyxml2.rb` for an example of a library formula with a good test. If not, no worries. Thanks!

Comment (tvon) [inline Library/Formula/pgcli.rb]: @mikemcquaid unless webmock or similar is available, `--help` is the best option available right now.

**Comment by tvon** [comment]
@tdsmith all set, should I squash this into one commit?

**Comment by tdsmith** [comment]
I got it; thanks.

**Comment by telemachus** [comment]
I'm curious: pgcli is available via `pip`. I thought [such items were discouraged from becoming independent formulas](https://github.com/Homebrew/homebrew/blob/master/share/doc/homebrew/Gems,-Eggs-and-Perl-Modules.md)?

**Comment by tdsmith** [comment]
The heuristic I'm applying is command-line applications that we can sandbox away from the rest of the system are generally fine; libraries are generally not (at least in the absence of a compelling use case).

**Comment by amjith** [comment]
@telemachus Someone else had asked a similar question on HN when I first requested help from the community to make a brew formula. 

When I read through the [guidelines](https://github.com/Homebrew/homebrew/blob/master/share/doc/homebrew/Python-for-Formula-Authors.md), they made it quite clear that applications written in Python acceptable, but libraries are not.  

To quote the doc: 

> Homebrew is happy to accept applications that are built in Python, whether the apps are available from PyPI or not. Homebrew generally won't accept libraries that can be installed correctly with pip install foo. 

`pgcli` falls in the application category.

**Comment by telemachus** [comment]
That's interesting. I wasn't aware of the distinction between apps and libraries that you both cite.

Thanks for explaining.

**Pull request closed without merging**